### PR TITLE
Move AggregationNode to SPI

### PR DESCRIPTION
**Pull request by sachdevs**
Follow up after #13404, depends on that one first.
In preparation for plan pushdown, move Aggregation plan node to SPI.

```
== RELEASE NOTES ==
General 
* Move `AggregationNode` to SPI. Connectors can now pushdown aggregation to table scan.
```

### CODE REVIEW

**Comment by sachdevs** [comment]
Rebased and cleaned as depending PR  #13404 was merged.

File: presto-spi/src/main/java/com/facebook/presto/spi/plan/AggregationNode.java [hunk 1]
```diff
@@ -246,17 +211,17 @@ public boolean isStreamable()
 
     public static GroupingSetDescriptor globalAggregation()
```
Comment (sachdevs) [inline presto-spi/src/main/java/com/facebook/presto/spi/plan/AggregationNode.java]: Do we want to keep GroupingSetDescriptor in SPI?

File: presto-spi/src/main/java/com/facebook/presto/spi/plan/AggregationNode.java [hunk 2]
```diff
@@ -68,7 +65,7 @@ public AggregationNode(
         super(id);
 
         this.source = source;
-        this.aggregations = ImmutableMap.copyOf(requireNonNull(aggregations, "aggregations is null"));
+        this.aggregations = unmodifiableMap(new HashMap<>(requireNonNull(aggregations, "aggregations is null")));
```
Comment (highker) [inline presto-spi/src/main/java/com/facebook/presto/spi/plan/AggregationNode.java]: Let's keep the order with linked hash map

File: presto-spi/src/main/java/com/facebook/presto/spi/plan/AggregationNode.java [hunk 3]
```diff
@@ -246,17 +211,17 @@ public boolean isStreamable()
 
     public static GroupingSetDescriptor globalAggregation()
     {
-        return singleGroupingSet(ImmutableList.of());
+        return singleGroupingSet(unmodifiableList(new ArrayList<>()));
```
Comment (highker) [inline presto-spi/src/main/java/com/facebook/presto/spi/plan/AggregationNode.java]: Use `emptyList`

File: presto-spi/src/main/java/com/facebook/presto/spi/plan/AggregationNode.java [hunk 4]
```diff
@@ -246,17 +211,17 @@ public boolean isStreamable()
...
     public static GroupingSetDescriptor singleGroupingSet(List<VariableReferenceExpression> groupingKeys)
     {
         Set<Integer> globalGroupingSets;
         if (groupingKeys.isEmpty()) {
-            globalGroupingSets = ImmutableSet.of(0);
+            globalGroupingSets = unmodifiableSet(Collections.singleton(0));
         }
         else {
-            globalGroupingSets = ImmutableSet.of();
+            globalGroupingSets = unmodifiableSet(new HashSet<>());
```
Comment (highker) [inline presto-spi/src/main/java/com/facebook/presto/spi/plan/AggregationNode.java]: `emptySet()`

File: presto-spi/src/main/java/com/facebook/presto/spi/plan/AggregationNode.java [hunk 5]
```diff
@@ -286,9 +251,9 @@ public GroupingSetDescriptor(
                 checkArgument(!globalGroupingSets.isEmpty(), "no grouping keys implies at least one global grouping set, but none provided");
             }
 
-            this.groupingKeys = ImmutableList.copyOf(groupingKeys);
+            this.groupingKeys = unmodifiableList(new ArrayList<>(groupingKeys));
             this.groupingSetCount = groupingSetCount;
-            this.globalGroupingSets = ImmutableSet.copyOf(globalGroupingSets);
+            this.globalGroupingSets = unmodifiableSet(new HashSet<>(globalGroupingSets));
```
Comment (highker) [inline presto-spi/src/main/java/com/facebook/presto/spi/plan/AggregationNode.java]: Same, let's keep the order

**Pull request merged**